extension/intents/pocket/pocketContentScript.js

```diff
@@ -0,0 +1,10 @@
+/* globals communicate */
+
+this.pocketContentScript = (function() {
+  communicate.register("save", async message => {
+    const url = message.url;
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "https://getpocket.com/save?url=" + url, true);
+    xhr.send();
```

Comment:
I'd prefer if this used [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) like:

```js
const resp = fetch(url);
if (resp.ok && !resp.redirected) {
  context.displayText("Saved");
}
```

Note also that quoting is important here, so use `"?url=" + encodeURIComponent(url)` – the current code will fail to save the query string of a page.